src/sidebar/store/create-store.js

```diff
@@ -7,6 +7,85 @@ import immutable from '../util/immutable';
 
 import { createReducer, bindSelectors } from './util';
 
+/**
+ * Helper that strips the first argument from a function type.
+ *
+ * @template F
+ * @typedef {F extends (x: any, ...args: infer P) => infer R ? (...args: P) => R : never} OmitFirstArg
+ */
+
+/**
+ * Helper that converts an object of selector functions, which take a `state`
+ * parameter plus zero or more arguments, into selector methods, with no `state` parameter.
+ *
+ * @template T
+ * @typedef {{ [K in keyof T]: OmitFirstArg<T[K]> }} SelectorMethods
+ */
+
+/**
+ * Map of action name to reducer function.
+ *
+ * @template State
+ * @typedef {{ [action: string]: (s: State, action: any) => Partial<State> }} Reducers
+ */
+
+/**
+ * Configuration for a store module.
+ *
+ * @template State
+ * @template {object} Actions
+ * @template {object} Selectors
+ * @template {object} RootSelectors
+ * @typedef Module
+ * @prop {(...args: any[]) => State} init -
+ *   Function that returns the initial state for the module
+ * @prop {string} namespace -
+ *   The key under which this module's state will live in the store's root state
+ * @prop {Reducers<State>} update -
+ *   Map of action types to "reducer" functions that process an action and return
+ *   the changes to the state
+ * @prop {Actions} actions
+ *   Object containing action creator functions
+ * @prop {Selectors} selectors
+ *   Object containing selector functions
+ * @prop {RootSelectors} [rootSelectors]
+ */
+
+/**
+ * Replace a type `T` with `Fallback` if `T` is `any`.
+ *
+ * Based on https://stackoverflow.com/a/61626123/434243.
+ *
+ * @template T
+ * @template Fallback
+ * @typedef {0 extends (1 & T) ? Fallback : T} DefaultIfAny
```

Comment:
For example:

`DefaultIfAny<string, number>` is `string`
`DefaultIfAny<any, number>` is `number`

This is used below to ensure correct handling of modules which don't define a `rootSelector` key.